src/adaptation/precond.jl

```diff
@@ -2,389 +2,227 @@
 #### Robust online (co-)variance estimators.
 ####
 
-## Variance estimator
+abstract type MassMatrixAdaptor <: AbstractAdaptor end
```

Comment:
It seems all subtypes of `MassMatrixAdaptor` have a type parameter `T` - would it make sense to include it already to `MassMatrixAdaptor` or even `AbstractAdaptor`?